### PR TITLE
Fix 1525

### LINKED ISSUE #1525: Crash when copy from cloud to internal memory
````
2018-11-23 02:49:49.147 10454-11135/com.amaze.filemanager E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #7
    Process: com.amaze.filemanager, PID: 10454
    java.lang.RuntimeException: An error occurred while executing doInBackground()
        at android.os.AsyncTask$3.done(AsyncTask.java:354)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: java.lang.IllegalStateException: This is not the first datapoint!
        at com.amaze.filemanager.asynchronous.services.AbstractProgressiveService.addFirstDatapoint(AbstractProgressiveService.java:183)
        at com.amaze.filemanager.asynchronous.services.CopyService$DoInBackground.doInBackground(CopyService.java:260)
        at com.amaze.filemanager.asynchronous.services.CopyService$DoInBackground.doInBackground(CopyService.java:231)
        at android.os.AsyncTask$2.call(AsyncTask.java:333)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:764) 
````

### PR DESCRIPTION
**Pull request by VishalNehra**
Fixes #1525

### CODE REVIEW

File: app/src/main/java/com/amaze/filemanager/asynchronous/services/CopyService.java [hunk 1]
```diff
@@ -257,7 +257,12 @@ protected Void doInBackground(Bundle... p1) {
 
             watcherUtil = new ServiceWatcherUtil(progressHandler);
 
-            addFirstDatapoint(sourceFiles.get(0).getName(), sourceFiles.size(), totalSize, move);
+            try {
+                addFirstDatapoint(sourceFiles.get(0).getName(), sourceFiles.size(), totalSize, move);
+            } catch (IllegalStateException e) {
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/asynchronous/services/CopyService.java]: This `Exception` shouldn't be catched, as it should never be thrown. The underlying bug should be fixed.

Comment (VishalNehra) [reply app/src/main/java/com/amaze/filemanager/asynchronous/services/CopyService.java]: The exception shouldn't be raised after this fix. I put this code for ease of testing. Can remove it if you want.

**Comment by VishalNehra** [comment]
@EmmanuelMess can this be included with today's release?

**Comment by EmmanuelMess** [comment]
@TranceLove Please check asap

**Comment by TranceLove** [comment]
@EmmanuelMess @VishalNehra sorry for late reply.

Tested on Oneplus 2 running Slim7 (7.1.2) with Amaze Cloud Plugin and Google Drive, previous symptom that Amaze FM crashed after copying file from cloud to local internal storage was gone after applying the changeset. I'd vote a +1.

**Pull request merged**